components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/util/MetricFormatter.java

```diff
@@ -94,6 +91,6 @@ public StringBuilder formatMetrics(Enumeration<MetricFamilySamples> metricsList)
             }
             metric.append(metricData);
         }
-        return metricsValue.append(metric);
```

Comment:
Shall we leave the existing implementation as it is, and return string by calling metricsValue.toString()?

Reply:
Fixed